### ISSUE TITLE
Enrich indices auto expand to all data nodes. Should we reduce the expansion?

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/elasticsearch#87753]
### Description

Enrich indices have by default 0-all.
On big cluster (e.g. 40 data nodes), it means 40 shards per Enrich policy.

Could we make it smarter and for example expand to all data nodes with the ingest role?
This might limit the expansion to the nodes where ingestion might actually happen.

**Comment by lucabelluccini** [comment]
I agree with the edge case with dedicated ingest. I thought to implement this only in case there are no ingest roles, but I agree for a distributed system is not the best.
Your suggestion to allow customizing the allocation tier preference is probably the best tradeoff.
Thank you for the quick triaging 🙇

**Comment by WaaZaa666** [comment]
I was redirected here by support. Would like to bump this issue up the priority list a bit. At this time we have a dozen enrich jobs, which each have there own dataset to enrich. We have hot, warm and frozen tier to store all our data. At this time we have too much data on our warm tier, and we could reduce the amount of data by simply changing the enrich jobs to only use the hot tier for storing the enrichment data. This cluster is not all that large, but with 10 warm nodes each storing a couple of 10's of gigabytes, the amount of storage which is needlessly used adds up. Only our _hot_ tier does indexing, so in my opinion it is not necessary to store all this data on the _warm_ tier.  

However, we cannot change the tier that enrich uses to store the data. It would be beneficial if we could just create index templates for .enrich, or that this is configurable in the enrich jobs. I would prefer the index template option, so we could also set other options on the .enrich indices in the future.
